### ISSUE TITLE
Pass a reason to onSessionEnded

### ISSUE THREAD

**Issue by avsdev-cw** [rstudio/shiny#3940]
I currently already use session$token and onSessionEnded to track when a user starts a session and leaves a session and thus how many users are currently using an application. (Useful for knowing if it is safe to restart an application or if I am going to interrupt a user doing work!). However, aside from looking at the output log, I am having issues finding a way to detect why a session has ended (more specifically, if the session has ended with an error).

Using the likes of rlang::last_error is too global - if one session ends with an error, rlang::last_error will be set for all future sessions as well. Less interested if an output has produced an error which has been captured and returned to the user, more interested in an error which causes the session to crash/end prematurely.

In short: can the reason a session has ended (e.g. user disconnect or error) be accessible or passed through to onSessionEnded?
